File: guides/dynamodb/src/main/java/example/micronaut/BooksController.java
package example.micronaut;

import io.micronaut.http.annotation.Controller;
import io.micronaut.http.annotation.Get;

import java.util.List;

@Controller("/books")
public class BooksController {

    private final BookRepository bookRepository;

    public BooksController(BookRepository bookRepository) {
        this.bookRepository = bookRepository;
    }

    @Get
    public List<Book> index() {
        return bookRepository.findAll();
    }
}

File: guides/micronaut-oauth2-client-credentials-oracle-identity-domain/bookcatalogue/python/src/example/micronaut/books_controller.py
from micronaut.http.annotation import Get
from micronaut.security.annotation import Secured
from micronaut.security.rules import SecurityRule

from .book import Book


@Get("/books")
@Secured(SecurityRule.IS_AUTHENTICATED)  # <1>
def index() -> list[Book]:
    building_microservices = Book("1491950358", "Building Microservices")
    release_it = Book("1680502395", "Release It!")
    continuous_delivery = Book("0321601912", "Continuous Delivery:")
    return [building_microservices, release_it, continuous_delivery]
